pkg/backends/influxdb/backend.go

```diff
@@ -142,17 +143,35 @@ func (b *Backend) getMetricPoints(metric *healthautoexport.Metric, tags []lp.Tag
 		points = append(points, point)
 	}
 
+	// Add points for detailed sleep analysis.
 	for _, s := range metric.SleepAnalyses {
-		points = append(points, makeSleepPoint(SleepAnalysisDetailed, s.Source, s.Value, 1, nil, s.StartDate, tags))
+		points = append(points, makeSleepPoint(MeasurementSleepAnalysisDetailed, s.Source, s.Value, 1, nil, s.StartDate, tags))
 		// end point has state = 0 (off)
-		points = append(points, makeSleepPoint(SleepAnalysisDetailed, s.Source, s.Value, 0, &s.Qty, s.EndDate, tags))
+		points = append(points, makeSleepPoint(MeasurementSleepAnalysisDetailed, s.Source, s.Value, 0, &s.Qty, s.EndDate, tags))
 	}
 
+	// Add points for aggregated sleep analysis.
 	for _, a := range metric.AggregatedSleepAnalyses {
-		points = append(points, makeSleepPoint(SleepAnalysisAggregated, a.SleepSource, "asleep", 1, nil, a.SleepStart, tags))
-		points = append(points, makeSleepPoint(SleepAnalysisAggregated, a.SleepSource, "asleep", 0, &a.Asleep, a.SleepEnd, tags))
-		points = append(points, makeSleepPoint(SleepAnalysisAggregated, a.InBedSource, "inBed", 1, nil, a.InBedStart, tags))
-		points = append(points, makeSleepPoint(SleepAnalysisAggregated, a.InBedSource, "inBed", 0, &a.InBed, a.InBedEnd, tags))
+		// Support old aggregated sleep analysis format prior to HAE v6.6.2.
+		if a.SleepSource != "" {
+			points = append(points, makeSleepPoint(MeasurementSleepAnalysisAggregated, a.SleepSource, "asleep", 1, nil, a.SleepStart, tags))
+			points = append(points, makeSleepPoint(MeasurementSleepAnalysisAggregated, a.SleepSource, "asleep", 0, &a.Asleep, a.SleepEnd, tags))
+		}
+		if a.InBedSource != "" {
+			points = append(points, makeSleepPoint(MeasurementSleepAnalysisAggregated, a.InBedSource, "inBed", 1, nil, a.InBedStart, tags))
+			points = append(points, makeSleepPoint(MeasurementSleepAnalysisAggregated, a.InBedSource, "inBed", 0, &a.InBed, a.InBedEnd, tags))
+		}
```

Comment:
Apologies for the sparse replies, following up on this now.

> My best guess is that `sleepStart`/`sleepEnd` has changed definitions.

I think this is the case, since `inBed` is now considered one of the possible phases during sleep.

> I was thinking that we should track these 2 metrics as InBed was previously tracked

Are you referring to using 1/0 to track the start/end of inBed? In this case, since we only have sleepStart/sleepEnd to play with, I think there's no way to retain this original behaviour.

Reply:
Correct, and I agree.